File: ScribBox/app/src/main/java/com/example/scribbox/MainActivity.java
package com.example.scribbox;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class MainActivity extends AppCompatActivity {
    private Button logIn,SignUp;
    private EditText logInMail, loginPassword;
    private TextView ForgotPassword;

    private FirebaseAuth firebaseAuth;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        getSupportActionBar().hide();
        firebaseAuth = FirebaseAuth.getInstance();
        FirebaseUser firebaseUser = firebaseAuth.getCurrentUser();

        /**
         * If user is already logged In then no need to login again and again
          */


        if(firebaseUser!=null){
            finish();
            startActivity(new Intent(MainActivity.this,NotesActivity.class));
        }

        /**
         * Finding all the views
         */

        logIn = findViewById(R.id.btnlogIn);
        SignUp = findViewById(R.id.btnSignUp);
        logInMail = findViewById(R.id.LoginMail);
        loginPassword = findViewById(R.id.LoginPassword);
        ForgotPassword = findViewById(R.id.forgotPasswordBtn);

        /**
         * On clicking signUp Button, then it will intent_toSignUpActivity"
         */

        SignUp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent_toSignUpActivity = new Intent(MainActivity.this, SignUp_Activity.class);
                startActivity(intent_toSignUpActivity);

            }
        });

        /**
         * On clicking Forgot Password Button, then it will intent_toForgetPasswordActivity"
         */

        ForgotPassword.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent_toForgetPasswordActivity = new Intent(MainActivity.this,ForgotPasswordActivity.class);
                startActivity(intent_toForgetPasswordActivity);
            }
        });

        logIn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String retrieve_LogInMail = logInMail.getText().toString().trim();
                String retrieve_LogInPassword = loginPassword.getText().toString().trim();

                if(retrieve_LogInMail.isEmpty() || retrieve_LogInPassword.isEmpty()){

                    Toast.makeText(MainActivity.this, "Provide Credentials", Toast.LENGTH_SHORT).show();

                }
                else{

                    /**
                     *  User login and OnComplete Listener
                     */

                     firebaseAuth.signInWithEmailAndPassword(retrieve_LogInMail,retrieve_LogInPassword).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                         @Override
                         public void onComplete(@NonNull Task<AuthResult> task) {
                                if(task.isSuccessful()){
                                    checkEmailVerification();
                                }
                                else {
                                    Toast.makeText(MainActivity.this, "Account Doesn't Exist", Toast.LENGTH_SHORT).show();
                                    firebaseAuth.signOut();
                                }
                         }
                     });

                }

            }
        });


    }

    private void checkEmailVerification() {

        FirebaseUser firebaseUser = firebaseAuth.getCurrentUser();
        if(firebaseUser.isEmailVerified() == true){
            Toast.makeText(this, "Successfully Logged In", Toast.LENGTH_SHORT).show();
            finish();
            startActivity(new Intent(MainActivity.this,NotesActivity.class));
        }else{
            Toast.makeText(this, "You haven't verified Email", Toast.LENGTH_SHORT).show();
        }

    }


}

File: FirebaseTtutorial/app/src/main/java/com/example/firebasettutorial/MainActivity.kt
package com.example.firebasettutorial

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener

class MainActivity : AppCompatActivity() {
    lateinit var database: FirebaseDatabase
    lateinit var ref: DatabaseReference
    lateinit var name: EditText
    lateinit var age: EditText
    lateinit var getName: TextView
    lateinit var getAge: TextView
    lateinit var pushData: Button
    lateinit var getData: Button
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        database = FirebaseDatabase.getInstance()
        ref = database.reference

        name = findViewById(R.id.name)
        age = findViewById(R.id.age)
        getName = findViewById(R.id.getName)
        getAge = findViewById(R.id.getAge)
        pushData = findViewById(R.id.pushData)
        getData = findViewById(R.id.getData)


        pushData.setOnClickListener {

            val name = name.text.toString()
            val age = age.text.toString()
            ref.child("name").setValue(name)
            ref.child("age").setValue(age)
        }

        getData.setOnClickListener {

            ref.addListenerForSingleValueEvent(object : ValueEventListener {
                override fun onDataChange(dataSnapshot: DataSnapshot) {
                    val name = dataSnapshot.child("name").value
                    val age = dataSnapshot.child("age").value

                    if (name != null) {
                        getName.text = "name is $name"
                    }

                    if (age != null) {
                        getAge.text = "age is $age years"
                    }

                }


//                val map = mutableMapOf<String, Any>()
//                for (child in dataSnapshot.children) {
//                    val key = child.key
//                    val value = child.value
//                    if (key != null && value != null) {
//                        map[key] = value
//                    }
//                }
//
//                val name = map["name"]
//                val age = map["age"]
//
//                if (name != null) {
//                    getName.text = "name is $name"
//                }
//
//                if (age != null) {
//                    getAge.text = "age is $age years"
//                }
//            }


                override fun onCancelled(databaseError: DatabaseError) {

                }
            })


        }
    }
}